wpengine/faustjs
# Merl 361 eslint config consolidation

## Description

- Took each eslintrc.js file in the Core, Next, and React packages and added their unique options to the root level eslintrc.js file.

- Took each devDependency from the package.json files in the Core, Next, and React packages and added their eslint or prettier devDependanciesto the root level package.json file’s devDependancies

- Removed .prettierrc.js files and eslintrc.js files from Core, Next, and React packages

-  Set root-level eslintrc.js to ignore paths including ‘test’ ‘when linting (**/test/*', and     '**/tests/*’)

- Fixed a single eslint error manually in the accessToken.ts file to ensure linting script was reflecting decrease in errors correctly

- Removed “"allowJs": true,” in order to squash “Typescript error: "Cannot write file ... because it would overwrite input file” error 
[reference for this removal](https://stackoverflow.com/questions/42609768/typescript-error-cannot-write-file-because-it-would-overwrite-input-file)



## Related Issue(s): n/a



## Testing

1. Checkout the MERL-361 branch
2. Run the build script 
3. Use the "lint-testing" script to see all package related linting issues (note: there will be 60 problems from Faustnx issues that are unrelated to the eslint configuration, and that will need to be examined in another ticket)
4. Please ensure no breaking changes



## Screenshots

When running "lint-testing" script, expect to see the following errors related to faustnx (that will require a separate ticket to clean!)
![60 problems (59 errors,](https://user-images.githubusercontent.com/50935135/181357414-b6a74f14-2d36-45c9-add4-b4f8fe572e61.png)



## Documentation Changes
n/a 


## Dependant PRs
n/a
